docs/source/using/managing-minishift.adoc

```diff
@@ -157,10 +160,67 @@ $ minishift config get memory
 To remove a persistent configuration option, you can use the
 link:../command-ref/minishift_config_unset{outfilesuffix}[`unset`] sub-command. For example:
 
+[source,sh]
 ----
 $ minishift config unset memory
 ----
 
+[[addons]]
+== Addons [WIP - Issue #177]
+
+Minishift allows to configure the vanilla OpenShift setup provided by cluster up via a so called
+addon mechanism. Addons are directories, containing a text file with the suffix _addon_. Each addon
+file contains some meta information as well as a list of command which will be run after OpenShift
+is provisoned. At the moment there are four different commands types:
+
+* Oc
+* OpenShift
+* Docker
+* Sleep
+
+=== Built-in addons
+
+Minishift provides a set f built-in addons which configure OpenShift in a more developer frinedly
+manner. To enable these built-in addons run:
+
+[source,sh]
+----
+$ minishift addons install --defaults
+----
+
+=== Enabling and disabling addons
+
+Addons are enabled resp. disable with the _addons enable_ resp. _addons disable_ commands.
+
+[source,sh]
+----
+$ minishift addons enable foo
+----
+
+[source,sh]
+----
+$ minishift addons disable foo
+----
+
+=== Providing custom addons
+
+To write a custom addons you need to provide at least on addon text file. This file needs to
+start with the following meta information:
+
+----
+# Name: anyuid
+# Description: Allows authenticated users to run images to run with USER as per Dockerfile
+----
+
+After the header you can provide commands which need match a minimalistic DSL. ...
```

Comment:
Also, needs step to install it. Is it? `minishift addons install <path to directory containing .addon file>